### PR TITLE
Update Mobile API for the new App HMI type WEB_VIEW

### LINKED ISSUE #254: [SDL 0273] WebEngine Projection mode
Proposal: [WebEngine Projection mode](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0273-webengine-projection-mode.md)

> This proposal is created based on discussion in https://github.com/smartdevicelink/sdl_evolution/issues/767.
It extends [SDl-0240 - WebEngine support for SDL JavaScript](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0240-sdl-js-pwa.md) to allow using the WebEngine app's web view for the HMI in addition to the other available SDL templates.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/912

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the revisions outlined in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/912#issuecomment-637749152).

> The proposal .md file was updated to reflect these revisions on 6/8/2020.

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Adds updates to API according to https://github.com/smartdevicelink/sdl_core/issues/3426

This PR is ready for review.
Summary

Adds:
 * New template WEB_VIEW
 * New App HMI type WEB_VIEW 
 * New exit reason "RESOURCE_CONSTRAINT"

CLA

 - [x]I have signed the CLA

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
Resolves https://github.com/smartdevicelink/rpc_spec/issues/254

File: MOBILE_API.xml [hunk 1]
```diff
@@ -449,6 +449,7 @@
         <element name="APP_UNAUTHORIZED" since="2.0" />
         <element name="PROTOCOL_VIOLATION" since="4.0" />
         <element name="UNSUPPORTED_HMI_RESOURCE" since="4.1" />
+        <element name="RESOURCE_CONSTRAINT" since="6.2" />
```
Comment (ghost) [inline MOBILE_API.xml]:
```suggestion
        <element name="RESOURCE_CONSTRAINT" since="7.0" />
```

File: MOBILE_API.xml [hunk 2]
```diff
@@ -2645,6 +2646,7 @@
         <element name="SYSTEM" />
         <element name="PROJECTION" since="4.5" />
         <element name="REMOTE_CONTROL" since="4.5" />
+        <element name="WEB_VIEW" since="6.2" />
```
Comment (ghost) [inline MOBILE_API.xml]:
```suggestion
        <element name="WEB_VIEW" since="7.0" />
```

File: MOBILE_API.xml [hunk 3]
```diff
@@ -2769,6 +2771,12 @@
                 Can be set as a root screen.
             </description>
         </element>
+        <element name="WEB_VIEW" rootscreen="true" since="6.2">
```
Comment (ghost) [inline MOBILE_API.xml]:
```suggestion
        <element name="WEB_VIEW" rootscreen="true" since="7.0">
```

**Review by ghost — CHANGES_REQUESTED** [review]
I like the description to the predefined layout. I only updated the version number as we jump to 7.0.

File: MOBILE_API.xml (hunk 1 above)
Comment (ghost) [inline MOBILE_API.xml]: Please add a description like "The application is unregistered due to hardware resource constraints. The system will shortly close the application to free up hardware resources." to this element.

File: README.md [hunk 4]
```diff
@@ -1327,6 +1327,7 @@ Enumeration listing possible app types.
 |`SYSTEM`||
 |`PROJECTION`||
 |`REMOTE_CONTROL`||
+|`WEB_VIEW`||
```
Comment (LuxoftAKutsan) [inline README.md:1330]: check if resource constraint should be in README

File: MOBILE_API.xml (hunk 1 above)
Comment (LuxoftAKutsan) [reply MOBILE_API.xml]: Comment added

File: README.md (hunk 4 above)
Comment (LuxoftAKutsan) [reply README.md:1330]: @kshala-ford fixed 58e96d5

**Comment by LuxoftAKutsan** [comment]
@kshala-ford all comments are processed

**Comment by SilvioDC-Ford** [comment]
Looking good now!
@theresalech this PR is ready for review and is approved by Ford.

**Comment by Jack-Byrne** [comment]
Changes look good, but I am waiting for approval of other webengine pr's before merging.

**Review by Jack-Byrne — APPROVED** [review]

**Pull request merged**